### PR TITLE
Optimizations for snabbwall & benchmarking script

### PR DESCRIPTION
**Pull request by takikawa**
This PR adds some optimizations and a benchmarking script.

For review purposes, I just wanted to make sure the optimizations seem reasonable and the benchmark sufficient. The dynasm-based commit is somewhat heavyweight, but it avoids a C wrapper that would complicate the build/linking.

The performance improvement is from ~1gbps before to ~4.5gbps after. (slightly faster if you benchmark only the `spy` command too)

CC @aperezdc, @dpino

### CODE REVIEW

File: src/apps/wall/scanner/ndpi_opt.dasl [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+   local mcode, size = Dst:build()
+   table.insert(__anchor, mcode)
+   if debug then
+      print("mcode dump: "..name)
+      dasm.dump(mcode, size)
+   end
+   return ffi.cast(prototype, mcode)
+end
+
+local function gen_nddpw(orig_f)
```
Comment (dpino) [inline src/apps/wall/scanner/ndpi_opt.dasl]: Possible typo: I wonder what nddp stands for, shouldn't be ndpp (ndpi_detection_process_packet)?

File: src/program/wall/tests/bench.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    # run the filter
+    ./snabb wall filter --cpu $CPU -p -e "{ BITTORRENT = 'drop', default = 'accept' }" -D $DURATION -4 192.168.0.1 -m "01:23:45:67:89:ab" intel10g $PCI1 > $output &
+    # blast with pcap traffic
+    ./snabb packetblaster replay -D $DURATION program/wall/tests/data/$1 $PCI2 > /dev/null
+    grep "bytes:.*packets:.*bps:" $output
+  done
+}
+
+benchmark BITTORRENT.pcap
+benchmark rtmp_sample.cap
```
Comment (dpino) [inline src/program/wall/tests/bench.sh:30]: I think the .pcap files are not included, are they? If they are too bug maybe is worth to provide from where download them.

**Review by dpino — APPROVED** [review]
Despite the nits, LGTM.

File: src/apps/wall/scanner/ndpi_opt.dasl (hunk 1 above)
Comment (takikawa) [reply src/apps/wall/scanner/ndpi_opt.dasl]: Thanks yeah, that's a typo.

File: src/program/wall/tests/bench.sh (hunk 2 above)
Comment (takikawa) [reply src/program/wall/tests/bench.sh:30]: The pcaps are already in the `src/program/wall/tests/data` folder (they're pcaps used for the functionality tests which were already there).

**Pull request merged**